src/main/scala/mesosphere/marathon/api/v2/PodsResource.scala

```diff
@@ -31,4 +50,207 @@ class PodsResource @Inject() (
+
+    withValid(id) { _ =>
+      val pod: PodDefinition = podSystem.find(id)
+      withAuthorization(ViewRunSpec, pod) {
+        ok(marshalJson(pod.asPodDef))
+      }
+    }(PodsValidation.idValidator)
+  }
+
+  @PUT @Timed @Path("""{id:.+}""")
+  def update(
+    @PathParam("id") id: String,
+    body: Array[Byte],
+    @DefaultValue("false")@QueryParam("force") force: Boolean,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(unmarshalJson[PodDef](decodeBytes(body, req))) { podDef =>
+
+      // podDef.id has just been validated, requiring equality for id validates it
+      require(id == podDef.id)
+      val pod = PodDefinition(withDefaults(podDef, podDefaults)).withCanonizedIds()
+
+      withAuthorization(UpdateRunSpec, pod) {
+
+        val (updatedPod, deploymentId) = podSystem.update(pod, force)
+        Events.maybePost(PodEvent(req.getRemoteAddr, req.getRequestURI, PodEvent.Updated))
+
+        val builder = Response
+          .ok(new URI(updatedPod.id.toString))
+          .entity(marshalJson(updatedPod.asPodDef))
+
+        deploymentId.foreach(id => builder.header(DEPLOYMENT_ID_HEADER, id))
+        builder.build()
+      }
+    }(createPodValidator and updatePodValidator)
+  }
+
+  @DELETE @Timed @Path("""{id:.+}""")
+  def remove(
+    @PathParam("id") id: String,
+    @DefaultValue("false")@QueryParam("force") force: Boolean,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(id) { _ =>
+      val pod: PodDefinition = podSystem.find(id)
+      withAuthorization(DeleteRunSpec, pod) {
+
+        val (deletedPod, deploymentId) = podSystem.delete(id, force)
+        Events.maybePost(PodEvent(req.getRemoteAddr, req.getRequestURI, PodEvent.Deleted))
+
+        val builder = Response
+          .ok(new URI(deletedPod.id.toString))
+          .entity(marshalJson(deletedPod.asPodDef))
+
+        deploymentId.foreach(id => builder.header(DEPLOYMENT_ID_HEADER, id))
+        builder.build()
+      }
+    }(PodsValidation.idValidator)
+  }
+
+  @GET @Timed @Path("""{id:.+}::status""")
+  def examine(
+    @PathParam("id") id: String,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(id) { _ =>
+      val pod: PodDefinition = podSystem.find(id)
+      withAuthorization(ViewRunSpec, pod) {
+        val status: PodStatus = podSystem.examine(id)
+        ok(marshalJson(status))
+      }
+    }(PodsValidation.idValidator)
+  }
+}
+
+/**
+  * public interfaces, specific to the pods api, that other pieces of the system implement and/or interact with
+  */
+object PodsResource {
+
+  /**
+    * Pod operations that reach this interface have already been authenticated, authorized, and otherwise validated.
+    * TODO(jdef) does it really make sense that deployment ID's are returned an Option[String] instead of String?
+    */
+  trait System {
```

Comment:
I actually like the "interface" defined here. An implemented by something else, somewhere else. It's the basis of the contract between the API and the underlying system. Plan was to mock this for unit tests.